docs/cartesian-product.adoc

```diff
@@ -0,0 +1,150 @@
+:page-title: Testing all test parameter combinations (Cartesian product)
+:page-description: Extends JUnit Jupiter with `@CartesianProductTest`, a special `@ParameterizedTest` which tests all combinations of its input
+
+Sometimes you want to test not only for specific inputs but also combinations of those inputs.
+`@CartesianProductTest` combines all test inputs and runs a test for each combination.
+
+[#_basic_use]
+== Basic Use
+
+`@CartesianProductTest` is used _instead_ of `@Test` or other such annotations (e.g. `@RepeatedTest`).
+The annotation can have a value (see <<Supplying a value to CartesianProductTest>>), or the test class can have a static factory method (see <<Writing a static factory method for the parameters>>) providing the arguments.
+
+== Supplying a value to CartesianProductTest
+
+If all your test parameters are strings, you can supply all input parameters simultaneously by giving a value to `@CartesianProductTest`.
```

Comment:
> giving a value 

With a single value there is no Cartesian product (or to be exactly: The CP only has one row). Would change it to plural, e.g. "by giving values to. These values are input for all parameters." or something like that.